cmake/AddResources.cmake

```diff
@@ -53,6 +53,13 @@ function(add_resources target)
 	foreach(file IN LISTS ARG_FILES)
 		target_sources(${target} PRIVATE "assets/${file}")
 		set_property(TARGET ${target} APPEND PROPERTY RESOURCE "assets/${file}")
+		get_filename_component(file_dir "${file}" DIRECTORY)
+		if(NOT "${file_dir}" STREQUAL "")
+			set(file_dir "/$file_dir")
+		endif()
+		set_source_files_properties("assets/${file}"
```

Comment:
Removed this bit then. I think that was the correct thing to do?

Reply:
Adding a file to the target’s `RESOURCE` property, and having a source file with the `MACOSX_PACKAGE_LOCATION` property, are mutually exclusive choices. If a file is in the `RESOURCE` list, `MACOSX_PACKAGE_LOCATION` is ignored and the file is dumped into the Resources directory.

`install(TARGET … RESOURCE DESTINATION)` (when it was working?!) is the same: the `RESOURCE` property is a flat list of files, and they all just get dumped into the destination as if you ran `install(FILES)` on that list.

So if this needs to support selecting individual files from subdirectories of the assets directory (which right now I don’t think it does?), then either the `RESOURCE` target property has to be abandoned totally and `MACOSX_PACKAGE_LOCATION` is used exclusively, or else `RESOURCE` has to be used only for files with no subdirectory and `MACOSX_PACKAGE_LOCATION` has to be used only for files that have subdirectories.

Hopefully this is clear now, because it sure took me about a week the first time to figure out WTF this thing is doing. :-)

Reply:
Okay, that makes reasonable amounts of sense, as far as CMake goes anyway. It seems there's a whole lot of ways to do installation, and they all seem to be broken in different ways. I should probably also pay attention to what I'm doing, cause I obviously just went and un-fixed top-level files not getting installed. Reverted that.